### ISSUE TITLE
Support manually setting CPU requests for dispatcher pods

### ISSUE THREAD

**Issue by lenalebt** [knative-sandbox/eventing-rabbitmq#823]
**Problem**
With release 1.5, CPU requests have been added to the dispatcher pods, but they are not configurable, which is an issue in some cases (see https://github.com/knative-sandbox/eventing-rabbitmq/issues/813#issuecomment-1158525701).

**[Persona:](https://github.com/knative/eventing/blob/main/docs/personas.md)**
System operator

**Exit Criteria**
CPU request for dispatcher pod can be set to any value, including removing it.

**Additional context (optional)**
If it's not an option to make this configurable, make sure only one dispatcher pod is needed per deployment consuming events. Currently, when creating a `Trigger` resource, one dispatcher pod is created. There are multiple options to solve this part:

* Supporting the new trigger filters (https://knative.dev/docs/eventing/experimental-features/new-trigger-filters/) which would allow to refactor multiple `Trigger` definitions into one
* Using multiple `Trigger` definitions from one dispatcher pod and consume multiple queues from one dispatcher.
* Probably more options that I don't know about :-)

**Comment by gab-satchi** [comment]
Hey @lenalebt, we implemented this recently and will be part of the 1.8 release. We've got a README for the new feature [here](https://github.com/knative-sandbox/eventing-rabbitmq/pull/948/files) and any feedback on it would be greatly appreciated.

One note on removing the requirements altogether: If any of the annotations are set, the defaulting will not take place. So setting the `limit` to 0 would make it unbounded and won't set a `request`. Ideally setting an empty annotation should remove it but that would throw an error as we use the Kubernetes resource quantity for validations.

**Comment by lenalebt** [comment]
Hey @gab-satchi ,
thanks! We sadly had quite a lot of trouble with the rabbitmq-broker and decided to move to Kafka a while ago, so I can't give you feedback with real-world experience here. But I'm pretty sure the resource allocation issues we had would have been solved this way!

**Comment by gab-satchi** [comment]
Thanks @lenalebt, 

If you recall any of the other troubling bits with rabbitmq-broker please let us know. We can work on improving it for current and future users.

**Comment by lenalebt** [comment]
I'll try to shortly remember, but I'm not very good at remembering things ;-).
* One of the troubles was it not recovering properly in many cases. Our development system runs on preemptible instances, which means they can basically die any moment, and we needed to manually care about getting it to work again on a daily basis. This works better with the kafka implementation (not trouble-free, but less overall trouble). I think many of those troubles have been resolved in the meantime (but were not at the time we switched over).
* I also remember that quorum queues were not supported at the time, so we could not easily spread queues over multiple nodes for failure tolerance. This put our services in the bad situation that we could not tolerate availability zone failures anymore, which the rest of the system was designed to handle.
* Having a separate container running per trigger was in general already a problem for us, because we're rather running many small services instead of less larger ones, so resource usage of one container multiplies. This is a design choice that I can understand wrt complexity in your service, but it was not a good choice for our implementation.
* The CPU configuration here was basically using way more resources than our actual services themselves.

Overall, those things add up. I was going for the RabbitMQ broker in the first place because I had some experience with RabbitMQ beforehand, and only had colleagues having a bad experience with Kafka (although that probably was in areas that we don't use, as we're not event-sourcing). The CPU thing was, in the end, the trigger to try the Kafka route.
